src/details/ArborX_DetailsTreeTraversal.hpp

```diff
@@ -86,41 +86,30 @@ struct TreeTraversal<BVH, Predicates, Callback, SpatialPredicateTag>
   {
     auto const &predicate = Access::get(predicates_, queryIndex);
 
-    Node const *stack[64];
-    Node const **stack_ptr = stack;
-    *stack_ptr++ = nullptr;
-    Node const *node = bvh_.getRoot();
+    Node const *node;
+    int next = 0; // start with root
```

Comment:
This should probably be fixed by using `getRoot()`. However, the code below is getting a bit more ugly with that change.